### ISSUE TITLE
on('connected') only happens once even after TWS restart

### ISSUE THREAD

**Issue by claude2** [pilwon/node-ib#65]
Situation:

1. Connect to the TWS API with script using `ib.connect()`
2. `on('connected')` handler is called
3. TWS exits, e.g. because of the daily logout or crash
4. User restarts TWS and logs in

The `on('connected')` handler will not be called again.

`on('disconnected')` is called correctly but I called ib.connect() from it on a 10 second interval and `on('connected')` was never called again.

How can we detect when TWS API is available again after disconnect?

**Comment by bfreis** [comment]
I believe this issue is related to one I was about to report -- so I'm adding a comment instead.

It seems like when the `net.Socket` emits an `'error'` event the `Socket` class from node-ib is simply propagating it. This is handle differently than the `'end'` event from `net.Socket` - the `'end'` event triggers the `Socket` class to set `this._connect = false`.

I identified this issue not with the daily logout (I run TWS continuously, fooling it so that it won't interrupt daily). I connect to TWS over an SSH tunnel. I noticed this issue first when my tunnel closed unexpectedly and an Express app I had with a persistent connection to TWS started misbehaving. The `ib` object emitted the 'disconnected' event, but it was impossible to connect again.

I believe both problems have the same root cause.

An easy fix I did is in the socket.js file change the handling of the net.Socket `'end'` event to `'close'`. That solved my problem. I didn't test much to see if this could cause other problems though...
